File: Problem5.ml
(*
 * 2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without 
 * any remainder.

 * What is the smallest positive number that is evenly divisible by all of the numbers from 1 
 * to 20?
*)

let rec checkDivis (n, divis) =
	if divis == 20 then (n, 1) else
		(if n mod divis == 0 then checkDivis(n, divis + 1) else (n, 0))

let rec findSmallest n =
	let (smallest, check) = checkDivis (n, 2) in
		if check == 1 then smallest else findSmallest n + 1

File: Problem5.py
#!/usr/bin/env python

# 2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without 
# any remainder.

# What is the smallest positive number that is evenly divisible by all of the numbers from 1 
# to 20?

import math

def checkDivisibility(num):
	for div in range(2, 21):
		if num % div != 0:
			return False
			
	return True

def findSmallestNumber(initial):
	b = initial * 10000000
	while initial < b:
		divisible = checkDivisibility(initial)
		if divisible == True:
			print initial
			break
		initial += 2
	print "Final: " + str(initial)
	

findSmallestNumber(100)
